src/GraphQL/Execution/ExecutionStrategy.cs

```diff
@@ -236,6 +237,8 @@ protected virtual async Task ExecuteNodeAsync(ExecutionContext context, Executio
                 if (!(result is IDataLoaderResult))
                 {
                     CompleteNode(context, node);
+                    // for non-dataloader nodes that completed without throwing an error, we can re-use the context
+                    System.Threading.Interlocked.Exchange(ref context.ReusableReadonlyResolveFieldContext, resolveContext);
```

Comment:
or

```suggestion
                    System.Threading.Interlocked.CompareExchange(ref context.ReusableReadonlyResolveFieldContext, resolveContext, null);
```
?

Although I think that the current version is also suitable, because in any case, the second instance of the context has already been created. We'll either replace it with our own or not, but one of them will go to garbage collection.